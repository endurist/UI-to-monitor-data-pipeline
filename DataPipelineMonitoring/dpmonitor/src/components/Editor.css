.editor-container {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.ace-editor {
    width: 40vw !important;
    min-width: 500px;
    height: 70vh !important;
    min-height: 500px;
}
