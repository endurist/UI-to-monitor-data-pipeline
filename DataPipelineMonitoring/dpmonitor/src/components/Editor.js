import React from 'react';
import AceEditor from 'react-ace';
import './Editor.css';

import 'ace-builds/src-noconflict/mode-python';
import 'ace-builds/src-noconflict/theme-solarized_dark';
import 'ace-builds/src-noconflict/theme-solarized_light';
import 'ace-builds/src-noconflict/theme-monokai';
import 'ace-builds/src-noconflict/theme-tomorrow';
import 'ace-builds/src-noconflict/ext-language_tools';

function Editor({ editorValue, updateEditorValue, readOnly, theme }) {
    function onChange(newValue) {
        updateEditorValue(newValue);
    }

    return (
        <div className='editor-container'>
            <AceEditor
                className='ace-editor'
                mode='python'
                onChange={onChange}
                theme={theme}
                value={editorValue}
                name='Script Editor'
                placeholder='Enter Script'
                fontSize={14}
                showPrintMargin={true}
                showGutter={true}
                highlightActiveLine={true}
                readOnly={readOnly}
                setOptions={{
                    enableBasicAutocompletion: true,
                    enableLiveAutocompletion: true,
                    showLineNumbers: true,
                    tabSize: 4,
                }}
            />
        </div>
    );
}

export default Editor;
