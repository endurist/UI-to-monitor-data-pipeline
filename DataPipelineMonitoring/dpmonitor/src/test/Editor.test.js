import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Editor from '../components/Editor';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';

afterEach(cleanup);

test('Editor window is rendered', () => {
    const { asFragment } = render(
        <Editor editorValue='example editor value' />
    );

    expect(
        asFragment(<Editor editorValue='example editor value' />)
    ).toMatchSnapshot();
});

test('Editor value should be equal to prop passed as value', () => {
    const wrapper = shallow(<Editor editorValue='test value' />);
    expect(wrapper.find('.ace-editor').at(0).props().value).toEqual(
        'test value'
    );
});

test('Editor placeholder is correct when empty', () => {
    const wrapper = shallow(<Editor editorValue='' />);
    expect(wrapper.find('.ace-editor').at(0).props().placeholder).toEqual(
        'Enter Script'
    );
});

test('Editor should be read only if passed readonly prop as true', () => {
    const wrapper = shallow(<Editor readOnly={true} />);
    expect(wrapper.find('.ace-editor').at(0).props().readOnly).toEqual(true);
});

test('Only one editor is rendered on the page', () => {
    const wrapper = shallow(
        <Editor editorValue='test value' readOnly={false} />
    );
    expect(wrapper.find('.ace-editor')).toHaveLength(1);
});

test('Two <Button /> components should be rendered', () => {
    const wrapper = shallow(
        <Editor editorValue='test value' readOnly={false} />
    );
    expect(wrapper.find(Button)).toHaveLength(2);
});

test('One <Form /> component should be rendered', () => {
    const wrapper = shallow(
        <Editor editorValue='test value' readOnly={false} />
    );
    expect(wrapper.find(Form)).toHaveLength(1);
});

test('The Form and Button texts should be rendered properly', () => {
    const { getByText } = render(
        <Editor editorValue='test value' readOnly={false} />
    );
    expect(getByText('Select Theme')).toBeInTheDocument();
    expect(getByText('Save Script')).toBeInTheDocument();
    expect(getByText('Schedule')).toBeInTheDocument();
});

test('Theme of the editor changes when the form option is clicked', () => {
    const container = render(<Editor editorValue='' readOnly={false} />);
    const formControl = container.getByTestId('theme-input');
    const optionSolarizedDark = container.getByTestId('solarized_dark');
    const optionTomorrow = container.getByTestId('tomorrow');

    expect(optionSolarizedDark.selected).toBe(true);
    expect(optionTomorrow.selected).toBe(false);

    fireEvent.change(formControl, { target: { value: 'tomorrow' } });

    expect(optionSolarizedDark.selected).toBe(false);
    expect(optionTomorrow.selected).toBe(true);
});
