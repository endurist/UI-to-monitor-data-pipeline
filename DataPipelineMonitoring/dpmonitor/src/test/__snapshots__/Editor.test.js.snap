// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor window is rendered 1`] = `
<DocumentFragment>
  <div
    class="editor-container"
  >
    <div
      class=" ace_editor ace_hidpi ace-solarized-dark ace_dark ace-editor"
      id="Script Editor"
      style="width: 500px; height: 500px; font-size: 14px;"
    >
      <textarea
        autocapitalize="off"
        autocorrect="off"
        class="ace_text-input"
        spellcheck="false"
        style="opacity: 0; font-size: 1px;"
        wrap="off"
      />
      <div
        aria-hidden="true"
        class="ace_gutter"
      >
        <div
          class="ace_layer ace_gutter-layer ace_folding-enabled"
          style="height: 1000000px;"
        />
      </div>
      <div
        class="ace_scroller"
        style="line-height: 0px;"
      >
        <div
          class="ace_content"
        >
          <div
            class="ace_layer ace_print-margin-layer"
          >
            <div
              class="ace_print-margin"
              style="left: 4px; visibility: visible;"
            />
          </div>
          <div
            class="ace_layer ace_marker-layer"
          />
          <div
            class="ace_layer ace_text-layer"
            style="height: 1000000px; margin: 0px 4px;"
          />
          <div
            class="ace_layer ace_marker-layer"
          />
          <div
            class="ace_layer ace_cursor-layer ace_hidden-cursors"
          >
            <div
              class="ace_cursor"
            />
          </div>
        </div>
      </div>
      <div
        class="ace_scrollbar ace_scrollbar-v"
        style="display: none; width: 20px;"
      >
        <div
          class="ace_scrollbar-inner"
          style="width: 20px;"
        >
           
        </div>
      </div>
      <div
        class="ace_scrollbar ace_scrollbar-h"
        style="display: none; height: 20px;"
      >
        <div
          class="ace_scrollbar-inner"
          style="height: 20px;"
        >
           
        </div>
      </div>
      <div
        style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: hidden;"
      >
        <div
          style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
        />
        <div
          style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
        >
          XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
        </div>
      </div>
    </div>
    <div
      class="options-container"
    >
      <form
        class="theme-form"
      >
        <div
          class="form-group"
        >
          <label
            class="theme-form-title form-label"
            for="selectTheme"
          >
            Select Theme
          </label>
          <select
            class="custom-select"
            data-testid="theme-input"
            id="selectTheme"
          >
            <option
              data-testid="solarized_dark"
              value="solarized_dark"
            >
              Solarized Dark
            </option>
            <option
              value="solarized_light"
            >
              Solarized Light
            </option>
            <option
              value="monokai"
            >
              Monokai
            </option>
            <option
              data-testid="tomorrow"
              value="tomorrow"
            >
              Tomorrow
            </option>
          </select>
        </div>
      </form>
      <button
        class="publish-button btn btn-outline-primary"
        type="button"
      >
        Save Script
      </button>
      <button
        class="schedule-button btn btn-outline-primary"
        type="button"
      >
        Schedule
      </button>
    </div>
  </div>
</DocumentFragment>
`;
